form.php

```diff
@@ -0,0 +1,72 @@
+<?php
+   $valid = array();
+   $errors = array();
+
+   if ($_POST) {
+     $subjectName = $_POST['subjectName'];
+     if (!$subjectName) {
+       $errors['subjectName'] = 'Името на предмета е задължително поле.';
+     } elseif (strlen($subjectName) > 150) {
+       $errors['subjectName'] = 'Името на предмета трябва да е по-малко от 150 символа.';
+     } else {
+       $valid['subjectName'] = $subjectName;
+     }
+
+     $lecturerName = $_POST['lecturerName'];
+     if (!$lecturerName) {
+       $errors['lecturerName'] = 'Името на лектора е задължително поле.';
+     } elseif (strlen($lecturerName) > 200) {
+       $errors['lecturerName'] = 'Името на лектора трябва да е по-малко от 200 символа.';
+     } else {
+       $valid['lecturerName'] = $lecturerName;
+     }
+
+     $description = $_POST['description'];
+     if (!$description) {
+       $errors['description'] = 'Описанието на предмета е задължително поле.';
+     } elseif (strlen($description) < 10) {
+       $errors['description'] = 'Описанието на предмета трябва да е по-голямо от 10 символа.';
+     } else {
+       $valid['description'] = $description;
+     }
+
+     $groups = $_POST['groups'];
+     if (!$groups) {
+       $errors['groups'] = 'Трябва да изберете поне една група.';
+     } else {
+       $valid['groups'] = $groups;
+     }
+
+     $credits = $_POST['credits'];
+     if (!$credits) {
+       $errors['credits'] = 'Кредит е задължително поле.';
+     } elseif (strlen($credits) < 0) {
+       $errors['credits'] = 'Кредита трябва да е положително число.';
+     } elseif (strlen($credits) > 20) {
```

Comment:
Please extract all constant values into constants/variables and give them meaningful names. This way you will know what is the purpose of the value when reading the code. For example here you would be able to reuse the value of `20` in the error message too.